### PR TITLE
display data without samples, provide description based on experiment type

### PR DESCRIPTION
**Pull request by wow-such-code**
needed for nanopore data model (unclassified reads files without sample!)
also renamed some variables for clarity

### CODE REVIEW

File: src/main/java/life/qbic/projectbrowser/controllers/DataHandler.java [hunk 1]
```diff
@@ -2669,6 +2669,48 @@ public String getSecondaryName(Sample samp, String datsetSecName) {
...
 
+  /**
+   * generates informative description from dataset secondary name and the dataset's experiment,
+   * given its ID. should only be used if a dataset does not have an associated sample
+   * 
+   * @param datasetSecondaryName
+   * @param experimentID
+   * @return
+   */
+  public String generateOrphanDatasetInformation(String datasetSecondaryName, String experimentID) {
```
Comment (christopher-mohr) [inline src/main/java/life/qbic/projectbrowser/controllers/DataHandler.java:2680]: It is not really orphan right?

File: src/main/java/life/qbic/projectbrowser/controllers/DataHandler.java [hunk 2]
```diff
@@ -2669,6 +2669,48 @@ public String getSecondaryName(Sample samp, String datsetSecName) {
...
+    if (datasetSecondaryName != null && datasetSecondaryName.isEmpty()) {
+      return datasetSecondaryName;
+    }
```
Comment (christopher-mohr) [inline src/main/java/life/qbic/projectbrowser/controllers/DataHandler.java:2683]: return empty String if it's `null`?

File: src/main/java/life/qbic/projectbrowser/controllers/DataHandler.java [hunk 3]
```diff
@@ -2669,6 +2669,48 @@ public String getSecondaryName(Sample samp, String datsetSecName) {
...
+    Experiment e = experiments.get(0);
+    Map<String, String> props = e.getProperties();
+    String type = e.getExperimentTypeCode();
+    switch (type) {
+      case "Q_NGS_NANOPORE_RUN":
+        String res = props.get("Q_FLOWCELL_BARCODE");
+        if (res == null) {
+          LOG.warn("No flow cell barcode was found for Nanopore Run " + experimentID
+              + ". Returning less informative metadata to display.");
+          res = "unclassified reads";
```
Comment (christopher-mohr) [inline src/main/java/life/qbic/projectbrowser/controllers/DataHandler.java:2699]: is that the term to use here?

File: src/main/java/life/qbic/projectbrowser/components/DatasetComponent.java [hunk 4]
```diff
@@ -247,45 +247,66 @@ public void updateUI(String type, String id) {
...
-                    SimpleDateFormat sd = new SimpleDateFormat("yyyy-MM-dd hh:mm a");
-                    String dateString = sd.format(date);
-                    String sampleID = samples.get(d.getCode());
-
-                    Sample dsSample = checkedTestSamples.get(sampleID);
-
-                    String secNameDS = d.getProperties().get("Q_SECONDARY_NAME");
-
-                    String secName = datahandler.getSecondaryName(dsSample, secNameDS);
+                String secNameDS = d.getProperties().get("Q_SECONDARY_NAME");
```
Comment (christopher-mohr) [inline src/main/java/life/qbic/projectbrowser/components/DatasetComponent.java:292]: Would be nicer to use an `enum` but I guess we did not do this in most of the cases unfortunately.

File: src/main/java/life/qbic/projectbrowser/controllers/DataHandler.java (hunk 1 above)
Comment (wow-such-code) [reply src/main/java/life/qbic/projectbrowser/controllers/DataHandler.java:2680]: true, maybe "composeDatasetWithoutSampleDescription"?

File: src/main/java/life/qbic/projectbrowser/components/DatasetComponent.java (hunk 4 above)
Comment (wow-such-code) [reply src/main/java/life/qbic/projectbrowser/components/DatasetComponent.java:292]: that's what I thought, we should do it in qportal 2.0 :)

File: src/main/java/life/qbic/projectbrowser/controllers/DataHandler.java (hunk 2 above)
Comment (wow-such-code) [reply src/main/java/life/qbic/projectbrowser/controllers/DataHandler.java:2683]: if it's null the experiment information is used, similar to the version for samples

File: src/main/java/life/qbic/projectbrowser/controllers/DataHandler.java (hunk 1 above)
Comment (christopher-mohr) [reply src/main/java/life/qbic/projectbrowser/controllers/DataHandler.java:2680]: `retrieveDatasetInfoWithoutSample` ?

Comment (wow-such-code) [reply src/main/java/life/qbic/projectbrowser/controllers/DataHandler.java:2680]: i like it

File: src/main/java/life/qbic/projectbrowser/controllers/DataHandler.java (hunk 2 above)
Comment (wow-such-code) [reply src/main/java/life/qbic/projectbrowser/controllers/DataHandler.java:2683]: ok, missing "!"

**Pull request merged**